picard/util/__init__.py

```diff
@@ -380,11 +380,17 @@ def iter_unique(seq):
 # order is important
 _tracknum_regexps = (
     # search for explicit track number (prefix "track")
-    r"track[\s_-]*(?:no|nr)?[\s_-]*(\d+)",
-    # search for 2-digit number at start of string
-    r"^(\d{2})\D",
-    # search for 2-digit number at end of string
-    r"\D(\d{2})$",
+    r"track[\s_-]*(?:(?:no|nr)\.?)?[\s_-]*(?P<number>\d+)",
```

Comment:
I wonder if it is worth the hassle, but perhaps we could compile regexps at this level:

```python
_tracknum_regexps = (
    # search for explicit track number (prefix "track")
    re.compile(r"track[\s_-]*(?:(?:no|nr)\.?)?[\s_-]*(?P<number>\d+)", re.I),
....

    for r in _tracknum_regexps:
        match = r.search(filename)
        if match:
....
```

Reply:
Actually yes, I already did this, but looks like I forgot to push it :)

Reply:
Please check https://github.com/metabrainz/picard/pull/1799/commits/b35acf62cf1afc55e8a37c935928171ff8b34b24

Reply:
Looks good :+1: